lib/RISCVBase.ml

```diff
@@ -286,18 +291,23 @@ type signed = Sign.t
 
 type 'k kinstruction =
   | INop
+  | Ret
   | OpI of opi * reg * reg * 'k
+  | OpA of opa * reg * lbl
   | OpIW of opiw * reg * reg * 'k
   | Op of op * reg * reg * reg
   | OpW of opw * reg * reg * reg
   | J of lbl
   | Bcc of cond * reg * reg * lbl
+  | MV of reg * reg
```

Comment:
`mv` is a pseudoinstruction (`mv rd,rs` being `addi rd,rs,0`). It would be natural not to have  specific abstract instruction. Instead one could parse `mv rd,rs` as the its expansion. Then one should probably print `addi rd,rs,0` as `mv rd,rs` and also emit the `mv` form as litmus output. However those last two steps are optional, although recommended.

I know that some other pseudoinstructions are not implemented that way (AArch64 `li r,k` for instance). However, it would be quite easy to do so in the case of RISCV `mv`, given that `addi` is already available.

Reply:
removed mv constructor, pretty print `add rd, rs, 0` as `mv rd,rs`, parse mv as add.